src/core/infer_request.cc

```diff
@@ -249,28 +249,49 @@ InferenceRequest::CopyAsNull(const InferenceRequest& from)
     new_input->SetData(data);
   }
 
-
   // Second pass
   size_t max_byte_size = 0;
+  size_t max_str_byte_size = 0;
   const std::string* max_input_name;
   for (const auto& input : from.OriginalInputs()) {
     // Skip shape tensors in this pass
     if (input.second.IsShapeTensor()) {
       continue;
     }
-    if (input.second.Data()->TotalByteSize() >= max_byte_size) {
-      max_byte_size = input.second.Data()->TotalByteSize();
-      max_input_name = &(input.first);
+
+    if (input.second.DType() == inference::DataType::TYPE_STRING) {
+      int64_t element_count = GetElementCount(input.second.Shape());
+
+      size_t str_byte_size = static_cast<size_t>(4 * element_count);
+      if (str_byte_size > max_byte_size) {
```

Comment:
You need to unconditionally use "max_str_byte_size = max(str_byte_size, max_str_byte_size)" because this could be a bigger string tensor but not be bigger than max_byte_size.

Reply:
Good catch, sorry about this. I fixed this in the next commit and tested this with the config below:

```
input [
  {
    name: "INPUT0"
    data_type: TYPE_STRING
    dims: [1]
  },
  {
    name: "INPUT1"
    data_type: TYPE_FP32
    dims: [7]
  },
  {
    name: "INPUT2"
    data_type: TYPE_STRING
    dims: [2]
  },
  {
    name: "INPUT3"
    data_type: TYPE_STRING
    dims: [2, 2]
  },
  {
    name: "INPUT4"
    data_type: TYPE_STRING
    dims: [2]
  }
]

output [
  {
    name: "OUTPUT0"
    data_type: TYPE_STRING
    dims: [-1]
  }
]

instance_group [{ kind: KIND_CPU }]
```

I hope it covers all the corner cases and there are not any further issues.